api-server/api-server-common/src/storage/impls/postgres/queries.rs

```diff
@@ -820,6 +820,33 @@ impl<'a, 'b> QueryFromConnection<'a, 'b> {
         Ok(())
     }
 
+    pub async fn get_pool_block_stats(
+        &self,
+        pool_id: PoolId,
+        chain_config: &ChainConfig,
+    ) -> Result<Option<u64>, ApiServerStorageError> {
```

Comment:
`get_pool_block_stats` sounds like a fairly generic function but it only returns the block count. Probably better for it to return some sort of `PoolBlockStats` struct, even if it only contains the count field for now so it can be extended in the future. (Alternatively just rename to `get_pool_block_count`)